mi/dataset/parser/common_regexes.py

```diff
@@ -37,6 +37,12 @@
 # A regex used to match a date in the format YYYY/MM/DD
 DATE_YYYY_MM_DD_REGEX = r'(\d{4})/(\d{2})/(\d{2})'
 
+# A regex used to match a date in the format YYYY-MM-DD
```

Comment:
Why not Zoidberg?
```python
r'(\d{4})[-\/]?(\d{2})[-\/]?(\d{2})'
```
This matches all three cases `YYYY/MM/DD, YYYY-MM-DD, YYYYMMDD`

Reply:
I made the change to DATE2_YYYY_MM_DD_REGX and added a '?' on the end to also match YYYY[-/]MM date formats as well.  That simplified the code a little and gave more options to the user for date formats.

I do not want to make the change to DATE_YYYY_MM_DD_REGEX as that is used in many places and I do not know what the impact would be by allowing additional date formats to match where they didn't before.  And I wouldn't have the time to regression test all the changes under this ticket.